exec/java-exec/src/main/java/org/apache/drill/exec/physical/impl/unorderedreceiver/UnorderedReceiverBatch.java

```diff
@@ -182,13 +184,18 @@ public IterOutcome next() {
         return IterOutcome.OUT_OF_MEMORY;
       }
 
+      // Transfer the ownership of this raw-batch to this operator for proper memory statistics reporting
+      batch = batch.transferBodyOwnership(oContext.getAllocator());
```

Comment:
@sachouche  Thank you for the changes. I was just contemplating if the below change
batch = batch.transferBodyOwnership(oContext.getAllocator()); 
needs to be moved before the check at line number 183.

Reply:
@HanumathRao,

- This will not affect the outcome since the unordered-receiver is a child of the context allocator (the minor fragment)
- Doing the check earlier seemed cleaner as there is no point of changing ownership when there is already an out-of-memory condition
- I think the code was written this way since implicitly, the network handlers are receiving the batch and then performing a change of ownership (from RPC to context allocator); this step could lead to an out-of-memory condition at the fragment level
- If your point is about reporting, that is attributing the OOM condition to the child operator, then I guess I don't have a problem with that.
@parthchandra what do you think?